src/main/webapp/WEB-INF/tags/instructor/results/questionPanel.tag

```diff
@@ -11,89 +11,89 @@
+      </c:otherwise>
+    </c:choose>
+  </div>
+  <div <c:if test="${questionPanel.collapsible}">class="panel-collapse collapse"</c:if>>
+    <div class="panel-body padding-0" <c:if test="${questionIndex != null}">id="questionBody-${questionIndex}"</c:if>>
 
-            <c:if test="${questionPanel.hasResponses}">
-                <div class="resultStatistics">
-                    ${questionPanel.questionStatisticsTable}
-                </div>
-                <c:if test="${questionPanel.showResponseRows}">
-                    <div class="table-responsive">
-                        <table class="table fixed-table-layout table-striped table-bordered dataTable margin-0">
-                            <thead class="background-color-medium-gray text-color-gray font-weight-normal">
-                                <tr>
-                                    <c:forEach items="${questionPanel.columns}" var="thElement">
-                                        <th ${thElement.attributesToString}>
-                                            ${thElement.content}
-                                            <c:if test="${questionPanel.isColumnSortable[thElement.content]}"><span class="icon-sort unsorted"></span></c:if>
-                                        </th>
-                                    </c:forEach>
-                                </tr>
-                            <thead>
-                            <tbody>
-                                <c:forEach items="${questionPanel.responses}" var="responseRow">
-                                    <results:responseRow responseRow="${responseRow}"/>
-                                </c:forEach>
-                            </tbody>
-                        </table>
-                    </div>
-                </c:if>
-            </c:if>
+      <c:if test="${!questionPanel.hasResponses}">
+        <div class="col-sm-12 no-response">
+          <i class="text-muted">There are no responses for this question or you may not have the permission to see the response</i>
+        </div>
+      </c:if>
 
+      <c:if test="${questionPanel.hasResponses}">
+        <div class="resultStatistics">
+          ${questionPanel.questionStatisticsTable}
         </div>
+        <c:if test="${questionPanel.showResponseRows}">
+          <div class="table-responsive">
+            <table class="table fixed-table-layout table-striped table-bordered dataTable margin-0">
+              <thead class="background-color-medium-gray text-color-gray font-weight-normal">
+                <tr>
+                  <c:forEach items="${questionPanel.columns}" var="thElement">
+                    <th ${thElement.attributesToString}>
+                      ${thElement.content}
+                      <c:if test="${questionPanel.isColumnSortable[thElement.content]}"><span class="icon-sort unsorted"></span></c:if>
+                    </th>
+                  </c:forEach>
+                </tr>
+              <thead>
```

Comment:
This looks like it should have been `</thead>` instead (not sure how the old code lasted this long without breaking). However, changing this might cause this file to be different from the test HTML files.